src/kili/services/plugins/upload.py

```diff
@@ -102,20 +157,42 @@ def _retrieve_upload_url(self, is_updating_plugin: bool) -> str:
         else:
             result = self.auth.client.execute(GQL_CREATE_PLUGIN, variables)
 
-        check_errors_plugin_upload(result, self.file_path, self.plugin_name)
+        check_errors_plugin_upload(result, self.plugin_path, self.plugin_name)
         upload_url = format_result("data", result)
         return upload_url
 
+    @staticmethod
+    def _create_zip(script_path: Path, requirements_path: Optional[Path], tmp_directory: Path):
+        """
+        Create a zip file from python file and requirements.txt
+        (if user has defined a path to a requirements.txt)
+        """
+        zip_path = tmp_directory / "archive.zip"
+
+        with ZipFile(zip_path, "w") as archive:
+            archive.write(script_path, "main.py")
+            if requirements_path:
+                archive.write(requirements_path)
```

Comment:
Shouldn't this also have a second argument "requirements.txt" as does the call for "main.py" ?

Reply:
I did it for `main.py` because if the user choose a path to a python file, the file could be named differently than `main.py`. While here `requirements.txt` could only be named `requirements.txt`